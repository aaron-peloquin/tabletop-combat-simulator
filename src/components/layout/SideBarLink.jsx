import { ListItem, ListItemIcon, ListItemText  } from "@material-ui/core"
import Link from "next/link"
import { withStyles } from "@material-ui/core/styles"

/**
 * Component for links as list items
 * @param {obj} props
    * @param {str} classes MUI object for CSS
    * @param {str} text The text of this link
    * @param {str} url The href
    * @param {jsx} children Optional icon
    * @return {jsx} <Link /> component intended to be placed inside of an MUI <ListItem />
 */
const SideBarLink = (props) => {
  const {text, url, children, classes} = props
  return <Link href={url}>
    <a className={classes.link}>
      <ListItem button>
        {children?<ListItemIcon>{children}</ListItemIcon>:""}
        <ListItemText primary={text} />
      </ListItem>
    </a>
  </Link>
}

/** MUI Styles for <SideBarLink /> */
const styles = {
  link: { textDecoration: "none", },
}

export default withStyles(styles)(SideBarLink)