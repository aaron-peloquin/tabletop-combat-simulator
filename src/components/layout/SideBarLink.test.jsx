import { Provider } from "react-redux"
import { createShallow } from "@material-ui/core/test-utils"
import toJson from "enzyme-to-json"

import SideBarLink from "./SideBarLink"

describe("<SideBarLink />", ()=>{
  const RenderShallowUntilComponent = createShallow({"untilSelector":"SideBarLink"})
  let store, props, Component
  beforeEach(async ()=>{
    props = {
      text: "hello",
      url: "/world",
    }

    /** Attach spy to store.dispatch */
    Component = RenderShallowUntilComponent(<Provider store={store}><SideBarLink {...props} /></Provider>)
  })

  it("loads", async () => {
    expect(typeof Component).toBe("object")
  })

  it("renders with correct text", () => {
    expect(Component.find("WithStyles(ListItemText)").prop("primary")).toBe(props.text)
  })

  it("renders with correct url", () => {
    expect(Component.find("Link").prop("href")).toBe(props.url)
  })

  it("snapshots", () => {
    expect(toJson(Component)).toMatchSnapshot()
  })
})