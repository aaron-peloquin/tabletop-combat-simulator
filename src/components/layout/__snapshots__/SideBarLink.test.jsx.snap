// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBarLink /> snapshots 1`] = `
<Link
  href="/world"
>
  <a
    className="SideBarLink-link-1"
  >
    <WithStyles(ListItem)
      button={true}
    >
      <WithStyles(ListItemText)
        primary="hello"
      />
    </WithStyles(ListItem)>
  </a>
</Link>
`;
